test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -162,6 +172,23 @@ TEST_P(ProxyProtocolTest, Basic) {
   disconnect();
 }
 
```

Comment:
Could you add this test for PPv1's minimal value?  I tested it and this works with your implementation, but I just want to make sure it keeps working even with future edits.  The concern is that the minimal PPv1 is just 15 bytes while the minimum for PPv2 is 16 bytes (LOCAL, no addr).  Want to make sure we never require >= 16 bytes on accident and break PPv1.

This is the PPv1 spec talking about `UNKNOWN`:
```
If the announced transport protocol is "UNKNOWN", then the receiver knows that
the sender speaks the correct PROXY protocol with the appropriate version, and
SHOULD accept the connection and use the real connection's parameters as if
there were no PROXY protocol header on the wire.
```

```
+TEST_P(ProxyProtocolTest, v1Minimal) {
+  connect();
+  write("PROXY UNKNOWN\r\nmore data");
+
+  expectData("more data");
+
+  if (GetParam() == Envoy::Network::Address::IpVersion::v4) {
+    EXPECT_EQ(server_connection_->remoteAddress()->ip()->addressAsString(), "127.0.0.1");
+  } else {
+    EXPECT_EQ(server_connection_->remoteAddress()->ip()->addressAsString(), "::1");
+  }
+  EXPECT_FALSE(server_connection_->localAddressRestored());
+
+  disconnect();
+}
```